packages/devkit/src/utils/package-json.ts

```diff
@@ -129,22 +137,43 @@ export function addDependenciesToPackageJson(
         ...(json.dependencies || {}),
         ...filteredDependencies,
       };
+
       json.devDependencies = {
         ...(json.devDependencies || {}),
         ...filteredDevDependencies,
       };
+
       json.dependencies = sortObjectByKeys(json.dependencies);
       json.devDependencies = sortObjectByKeys(json.devDependencies);
 
       return json;
     });
+
     return (): void => {
       installPackagesTask(tree);
     };
   }
   return () => {};
 }
 
+/**
+ * @returns The the incoming dependencies that are higher than the existing verions
+ **/
+function removeLowerVersions(
```

Comment:
Added a new function to do this, mainly because the `updateExistingDependenciesVersion` is checking for deps vs devDeps (and vice versa) instead of just making sure the incoming vs existing deps are trimmed correctly.